src/Sarif.Multitool/TransformCommand.cs

```diff
@@ -58,5 +81,28 @@ public static int Run(TransformOptions transformOptions)
 
             return 0;
         }
+
+        private string SniffVersion(string sarifPath, int tokenCountLimit = int.MaxValue)
+        {
+            int tokenCount = 0;
+
+            // When we're unit-testing, we'll retrieve a string representation of the file contents and pass this to a stream reader instance.
+            TextReader textReader = _testing ? new StreamReader(new MemoryStream(Encoding.UTF8.GetBytes(_fileSystem.ReadAllText(sarifPath)))) : new StreamReader(sarifPath);
```

Comment:
>textReader [](start = 23, length = 10)

Long line; suggest breaking it:
```
TextReader textReader = _testing
    ? the-one-thing
    : the-other-thing;
```

Reply:
You don't need the `_testing` flag at all. The first branch of the ternary operator works whether you have a real or a fake file system:
```
TextReader textReader = new StreamReader(new MemoryStream(Encoding.UTF8.GetBytes(_fileSystem.ReadAllText(sarifPath))));
```
If you chose the conditional to avoid having to read a whole, potentially large, "real" log file into memory, that's worth a comment.

---
In reply to: [229510405](https://github.com/Microsoft/sarif-sdk/pull/1125#discussion_r229510405) [](ancestors = 229510405)